### PR TITLE
Add basic UI

### PR DESCRIPTION
**Pull request by Ibnballo1**
In this PR, I implemented the following:

- I created an app.rb file that will serve as the console app entry point with the following methods:
    - List all books.
    - List all people.
    - Create a person (teacher or student, not a plain Person).
    - Create a book.
    - Create a rental.
    - List all rentals for a given person id.

- I created a main.rb file that defines the entry point with the following functionality:
    - Present the user with a list of options to perform.
    - Lets users choose an option.
    - If needed, ask for parameters for the option.
    - Have a way to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+require_relative './book'
+require_relative './teacher'
+require_relative './student'
+require_relative './rental'
+require_relative './person'
+require 'json'
+require_relative './user_input'
+require_relative './user_output'
+
```
Comment (rloterh) [inline app.rb:9]: - [ ] OPTOINAL
I suggest you add more details of a book or a person so as to make the app flow more intuitive for users. See the highlighted parts of the screenshot below for the same:

**Your interactive app:**

image


**Expected:**
image

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    puts "####\nWelcome to the School Library Ruby App!\n####"
+    while user_input != '7'
+      puts "Choose a number from the options below:\n"
+      app_options
+      print 'Enter Option [number]: '
+      user_input = gets.chomp
+      puts "\n\n"
+      check_selection(user_input)
+    end
+    puts "Exiting, thanks for using this app!\n\n" if user_input == '7'
```
Comment (rloterh) [inline app.rb:31]: - [ ] Your app is throwing an error in an attempt to exit, using option 7, as can be seen in the screenshot below.
Kindly ensure to improve upon this aspect.

**Exit error:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Ibnballo, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Good job with implementation so far.


## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

**You can also consider:**
- Kindly make it stand out by using this [Microverse recommended template](https://github.com/microverseinc/readme-template/blob/master/README.md), and giving detailed steps where applicable, and properly structuring your ReadMe
- I think you should change the header title _Authors_ to _Author_ in your ReadMe, as this project is worked on by a sole contributor. 
Implementing the above change(s) will make your ReadMe/ repository stand out even better 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (Ibnballo1) [reply app.rb:9]: Thanks, @rloterh  but I don't seem to understand what you ask me to do

File: app.rb (hunk 2 above)
Comment (Ibnballo1) [reply app.rb:31]: As for this, I corrected it before requesting for review, I don't know why it is still showing

I will check again

File: app.rb (hunk 1 above)
Comment (rloterh) [reply app.rb:9]: Hi @Ibnballo1 
Printing the details of persons and books should be formatted like the output highlighted in green.

Cheers! 😃

File: app.rb (hunk 2 above)
Comment (Ibnballo1) [reply app.rb:31]: Fixed

File: app.rb (hunk 1 above)
Comment (Ibnballo1) [reply app.rb:9]: My code looks exactly like this, but I don't seem to get what you're asking for!

Comment (Ibnballo1) [reply app.rb:9]: My code work exactly as that too

**Review by t-yanick — APPROVED** [review]
Hi @Ibnballo1, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**